src/core/lib/layer.js

```diff
@@ -161,71 +169,56 @@ export default class Layer {
 
   // PROJECTION METHODS
 
-  /**
-   * Projects a point with current map state (lat, lon, zoom, pitch, bearing)
-   *
-   * Note: Position conversion is done in shader, so in many cases there is no need
-   * for this function
-   * @param {Array|TypedArray} lngLat - long and lat values
-   * @return {Array|TypedArray} - x, y coordinates
-   */
+  // Projects a point with current map state (lat, lon, zoom, pitch, bearing)
+  // TODO - need to be extended to work with multiple `views`
   project(lngLat) {
     const {viewport} = this.context;
-    assert(Array.isArray(lngLat), 'Layer.project needs [lng,lat]');
+    assert(Array.isArray(lngLat));
```

Comment:
Non-related: these projection methods do not produce the right result unless `coordinateSystem` is LNG_LAT.